### PR TITLE
Add correct selinux label at plugin socket directory

### LINKED ISSUE #73240: Unprivileged device plugins fails to start grpc server
**What happened**:
If the device plugin is an unprivileged container and selinux is enabled, it fails to start grpc server.

```
Jan 23 15:31:28 localhost journal[23606]: pluginSocksDir:  /var/lib/kubelet/device-plugins 
Jan 23 15:31:28 localhost audit[9297]: AVC avc:  denied  { write } for  pid=9297 comm="sampledeviceplu" name="device-plugins" dev="dm-2" ino=4990510 scontext=system_u:system_r:container_t:s0:c18,c144 tcontext=system_u:object_r:unlabeled_t:s0 tclass=dir permissive=0 
Jan 23 15:31:28 localhost journal[23606]: Start error:  listen unix /var/lib/kubelet/device-plugins/dp.1548237688: bind: permission denied
```

**What you expected to happen**:
Unprivileged device plugins also able to work

**How to reproduce it (as minimally and precisely as possible)**:
0. Enable selinux
1. Deploy a k8s cluster
2. Deploy a device plugin daemonset without `privileged: true` and plugin sock dir as "/var/lib/kubelet/device-plugins/"

Plugin will not be able to register

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by vikaschoudhary16**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This PR adds appropriate selinux label at plugin socket directory

**Which issue(s) this PR fixes**:

Fixes #73240 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Set selinux label at plugin socket directory
```
/area hw-accelerators
@kubernetes/sig-node-bugs @kubernetes/sig-node-pr-reviews @jeremyeder

### CODE REVIEW

File: pkg/kubelet/config/defaults.go [hunk 1]
```diff
@@ -26,4 +26,5 @@ const (
 	DefaultKubeletContainersDirName          = "containers"
 	DefaultKubeletPluginContainersDirName    = "plugin-containers"
 	DefaultKubeletPodResourcesDirName        = "pod-resources"
+	DefaultKubeletPluginsDirSELinuxLabel     = "system_u:object_r:container_file_t:s0"
```
Comment (tallclair) [inline pkg/kubelet/config/defaults.go]: nit: This is called a default, but is it configurable in any way? Should it just be `KubeletPluignsDirSELinuxLabel`?

Comment (tallclair) [inline pkg/kubelet/config/defaults.go]: Is this label documented anywhere? Should it be?

**Review by tallclair — COMMENTED** [review]
Looks good, but I'm wondering about how this is communicated to the user.

File: pkg/kubelet/config/defaults.go (hunk 1 above)
Comment (vikaschoudhary16) [reply pkg/kubelet/config/defaults.go]: @tallclair no, i think nowhere in k8s documentation.
Should it be? ---- Documentation never hurts. I am just not sure where would it be appropriate to mention about this. Any suggestions please?

Comment (tallclair) [reply pkg/kubelet/config/defaults.go]: Sounds good to me.

Comment (vikaschoudhary16) [reply pkg/kubelet/config/defaults.go]: Will update proposals post merge of this PR

File: pkg/util/selinux/selinux_linux.go [hunk 2]
```diff
@@ -50,3 +50,8 @@ func (_ *realSELinuxRunner) Getfilecon(path string) (string, error) {
 	}
 	return selinux.FileLabel(path)
 }
+
+// FileLabel returns the SELinux label for this path or returns an error.
```
Comment (booxter) [inline pkg/util/selinux/selinux_linux.go]: does this comment belong here? We call ```SetFileLabel``` here not ```FileLabel```; and the former doesn't return label or error, just error.

**Review by booxter — APPROVED** [review]
This probably resolves the issue for privileged containers, but some, like SR-IOV Intel device plugin, are not privileged, and it would be sad to upgrade its privileges just for this SELinux label. Do you have an idea how to tackle that in case of unprivileged containers?

**Comment by booxter** [comment]
> This probably resolves the issue for privileged containers, but some, like SR-IOV Intel device plugin, are not privileged, and it would be sad to upgrade its privileges just for this SELinux label. Do you have an idea how to tackle that in case of unprivileged containers?

Disregard this comment, I misinterpreted warnings emitted by this patch as a sign that SELinux labels are applied by plugin itself and not kubelet.

File: pkg/util/OWNERS [hunk 3]
```diff
@@ -0,0 +1,10 @@
+reviewers:
+  - pmorie
```
Comment (smarterclayton) [inline pkg/util/OWNERS]: Generally we wouldn't add owners in the same PR.  Would be better to split this out.

File: pkg/kubelet/cm/devicemanager/manager.go [hunk 4]
```diff
@@ -206,6 +207,12 @@ func (m *ManagerImpl) Start(activePods ActivePodsFunc, sourcesReady config.Sourc
 
 	socketPath := filepath.Join(m.socketdir, m.socketname)
 	os.MkdirAll(m.socketdir, 0755)
+	if selinux.SELinuxEnabled() {
+		err = selinux.SetFileLabel(m.socketdir, config.KubeletPluginsDirSELinuxLabel)
```
Comment (smarterclayton) [inline pkg/kubelet/cm/devicemanager/manager.go]: This should really be `if err := x; err != nil`.  Using this style means that a later change in the code could access err, which is not what we want.

**Comment by smarterclayton** [comment]
I don't see a second review after the last set of changes?  @kubernetes/sig-node-pr-reviews please take a look

File: pkg/util/OWNERS (hunk 3 above)
Comment (vikaschoudhary16) [reply pkg/util/OWNERS]: Actually @tallclair suggested for this. I am fine moving this out as well.

**Comment by vikaschoudhary16** [comment]
ping @smarterclayton @tallclair

**Comment by smarterclayton** [comment]
/approve

But needs review (it looks ok to me but I don’t want to be the sign off)

**Pull request merged**